torch/csrc/jit/script/exit_transforms.cpp

```diff
@@ -125,6 +161,48 @@ struct ExitTransformer {
     return match_values;
   }
 
+  ExitPair transformLoop(Node* node) {
+    LoopView loop(node);
+    Block* body = loop.bodyBlock();
+    auto exit_pair = transformExits(body);
+    // if we're not exiting to outside the loop we don't need to do any work.
+    if (getExitStatus(exit_pair) == ExitStatus::WONT) {
+      return exit_pair;
+    }
+
+    // if we are, we need to update the loop continue condition so that
+    // we exit the loop if we've hit an exit
+    WithInsertPoint insert(body);
+    auto new_if = graph_->insertNode(graph_->create(prim::If, 0));
+    new_if->addInput(exit_pair.hasExited());
+    new_if->addBlock()->registerOutput(false_val_);
+    new_if->addBlock()->registerOutput(loop.nextCond());
+    auto new_condition = new_if->addOutput()->setType(BoolType::get());
+    loop.bodyBlock()->eraseOutput(0);
+    loop.bodyBlock()->insertOutput(0, new_condition);
+
+    // we also need to propagate hasExited() and exitValues() outside the loop
+
+    // we didn't exit if we didn't enter the loop
+    node->addInput(false_val_);
+    body->addInput()->setType(BoolType::get());
```

Comment:
It's pretty hard to track what's going on in this function. Could you please add a description or an example?